pynucastro/rates/rate.py

```diff
@@ -395,6 +409,48 @@ def swap_protons(self):
         self.fname = None    # reset so it will be updated
         self._set_print_representation()
 
+    def reactant_count(self, n):
+        """Return the number of times nucleus n appears as a reactant
+        in the rate.  Use the stoichiometry dict if present.
+
+        Parameters
+        ----------
+        n : Nucleus
+            the nucleus appearing as a reactant
+
+        Returns
+        -------
+        float
+
+        """
+
+        c_reac = self.reactants.count(n)
+
+        if self.stoichiometry and c_reac > 0:
```

Comment:
we should check keys in reactant/product_count too, if we don't want to pass in redundant reactant/product: 1 in stoichiometry.

Reply:
indeed, I seem to have missed this when I changed how I was doing things.

Reply:
actually, now I am not sure what you are asking here

Reply:
ah okay.  I think that is doable now, since I've changed everything to just use those new helper functions.

Reply:
this is done now